main/src/cgeo/geocaching/maps/mapsforge/v6/caches/CachesBundle.java

```diff
@@ -220,6 +224,18 @@ public void invalidate(final Collection<String> geocodes) {
         if (liveOverlay != null) {
             liveOverlay.invalidate(geocodes);
         }
+        if (wpOverlay != null) {
+            // get waypoint IDs for geocodes and invalidate them
+            final Set<Geocache> baseCaches = DataStore.loadCaches(geocodes, LoadFlags.LOAD_WAYPOINTS);
```

Comment:
To me this looks like something the waypoint overlay should implement and provide as a method to call from here, what do you think?

Reply:
yes, you're right. See updated PR